src/index.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  Pocket,
+  Configuration,
+  RelayResponse,
+  RpcErrorResponse,
+  RelayHeaders,
+  RelayRequest,
+  typeGuard
+} from "pocket-js-core"
+import { TransactionSigner } from "./transaction-signer"
+import {AbstractSocketProvider} from 'web3-providers';
+
+/**
+ * Pocket Web3 Provider
+ * Sends Relays to a service node in the Pocket Network
+ * @param {string} activeBlockchain - Blockchain hash.
+ * @param {Configuration} configuration - Pocket Configuration.
+ * @param {TransactionSigner} transactionSigner - (optional) Transaction signer object.
+ */
+export class PocketProvider extends AbstractSocketProvider {
+  public readonly pocket: Pocket
+  public readonly transactionSigner?: TransactionSigner
+  public activeBlockchain: string
+  public isConnected = true
+
+  constructor(
+    activeBlockchain: string,
+    configuration: Configuration,
+    transactionSigner?: TransactionSigner
+  ) {
+    super(undefined, 40000)
+
+    this.pocket = new Pocket(configuration)
+    this.activeBlockchain = activeBlockchain
+    this.transactionSigner = transactionSigner
+  }
+  /**
+   *
+   * @method send
+   * @param {RelayRequest} relay - Object containing the relay request.
+   * @returns {RelayResponse} - Relay response object.
+   * @memberof PocketProvider
+   */
+  public async send(payload: {}): Promise<RelayResponse | RpcErrorResponse> {
+    // Check for the pocket instance
+    if (!typeGuard(this.pocket, Pocket)) {
```

Comment:
Check for this in the constructor of the provider, not here.

Reply:
this check is unnecessary since the pocket instance and configuration are mandatory, instantiating the pocket instance never fails